shell/platform/darwin/ios/ios_gl_render_target.h

```diff
@@ -40,8 +41,8 @@ class IOSGLRenderTarget {
 
  private:
   fml::scoped_nsobject<CAEAGLLayer> layer_;
-  fml::scoped_nsobject<EAGLContext> context_;
-  fml::scoped_nsobject<EAGLContext> resource_context_;
+  std::shared_ptr<IOSGLContext> onscreen_context_;
```

Comment:
I think this might be better as `fml::WeakPtr<IOSGLContext>`. We really don't want anyone adding to the ref count for this - it should just be a unique_ptr held by `FlutterView`, and `FlutterView` should also hold a `WeakPtrFactory<IOSGLContext>` to vend weak pointers for it.

I believe that should be safe as long as the weak ptr is created on the GPU thread.  @chinmaygarde does that sound right to you?

Reply:
Right, it should only be created, accessed and collected on the GPU task runner. Which reminds me, I haven't audited this for thread safety yet.

Reply:
So I was worried about whether we can tie lifetime of the IOSGLContext as tightly to the FlutterView as that. I guess we have guarantees that the IOSGLContext will only ever be used if the FlutterView is alive and well?

Reply:
I think that's fine - if there's no flutter view we shouldn't be trying to render anything to an onscreen GL context